.github/workflows/tests.yaml

```diff
@@ -62,6 +58,18 @@ jobs:
 
       - run: poetry check --lock
 
+  build:
+    name: Check build
+    runs-on: ubuntu-latest
+    if: needs.changes.outputs.src == 'true'
+    needs: lockfile
+    steps:
+      - uses: actions/checkout@v4
+
+      - run: pipx run build
+
+      - run: pipx run twine check --strict dist/*
```

Comment:
I am a bit conflicted about this. We should actually rely on build producing valid artifacts and not actually need twine here.

If we do these, we are better off just doing `pipx run poetry build`.

Reply:
This is intended to test the PEP 517/pyproject-hooks flow. We don't use `poetry build` here as we have plenty of builder tests, but our build flow special-cases/partially ignores `build-system.requires`.

This step is really here to ensure other spec-compliant tools can build us.